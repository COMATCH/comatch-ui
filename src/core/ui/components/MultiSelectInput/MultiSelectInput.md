## MultiSelectInput

An input similar to our `Select` input, allowing the user to select multiple options.

This component is our customized wrapper around the 'react-select' component.
@see https://github.com/JedWatson/react-select

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**className** | `String` | `null` | :x: | Additional class names
**disabled** | `Boolean` | `false` | :x: | 
**display** | `Enum('block', 'inline')` | `'block'` | :x: | 
**id** | `String` | `null` | :x: | HTML id attribute
**inputError** | `String` | `''` | :x: | 
**label** | `Union<Array[]<ReactNode>\|ReactNode\|String>` | `''` | :x: | 
**label<1>** | `Array[]<ReactNode>` |  | :x: | 
**label<2>** | `ReactNode` |  | :x: | 
**label<3>** | `String` |  | :x: | 
**name** | `String` |  | :white_check_mark: | 
**onBlur** | `Function` | `noop` | :x: | 
**onChange** | `Function` | `noop` | :x: | 
**onInputChange** | `Function` | `noop` | :x: | 
**options** | `Array[]<Shape>` |  | :white_check_mark: | 
**options[].label** | `Union<String\|Number\|Boolean>` |  | :x: | 
**options[].label<1>** | `String` |  | :x: | 
**options[].label<2>** | `Number` |  | :x: | 
**options[].label<3>** | `Boolean` |  | :x: | 
**options[].value** | `Union<String\|Number\|Boolean>` |  | :x: | 
**options[].value<1>** | `String` |  | :x: | 
**options[].value<2>** | `Number` |  | :x: | 
**options[].value<3>** | `Boolean` |  | :x: | 
**required** | `Boolean` | `false` | :x: | 

