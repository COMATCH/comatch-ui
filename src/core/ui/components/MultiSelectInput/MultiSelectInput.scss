@import '../../../styles/variables.scss';

.MultiSelectInput {
    font-size: $co-font-sm;
    display: inline-block;
    margin-bottom: $co-inputs-margin-bottom;

    // common styles
    .InputLabel,
    &.block,
    &__input {
        display: block;
    }

    // Elements
    &__input {
        // coloring
        background-color: $co-color-white;
        border: 1px solid $co-color-mid-gray;
        border-radius: 3px;
        box-shadow: 0 0 4px 0 rgba(1, 156, 185, 0), inset 0 0 2px 0 rgba(193, 193, 193, 0.5);
        // sizing & whitespace
        min-height: 40px;
        width: 100%;
        // transition effects
        transition: box-shadow 250ms ease-out;
        transform: translateZ(0);
        will-change: box-shadow;
        // reset browser defaults
        outline: 0 !important;
        z-index: 3;

        &:focus {
            box-shadow: 0 0 4px 0 rgba(1, 156, 185, 1), inset 0 0 2px 0 rgba(193, 193, 193, 0.5);
        }
    }

    // States
    &.has-error {
        .SelectInput__input {
            box-shadow: 0 0 4px 0 rgba(173, 25, 25, 1), inset 0 0 2px 0 rgba(193, 193, 193, 0.5);
        }
    }

    &.disabled {
        color: $co-color-gray;
    }
}
