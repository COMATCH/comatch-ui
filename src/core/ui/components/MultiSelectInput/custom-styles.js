import { palette, typography } from '../../../styles/variables';

/**
 * Custom styles to adjust the react-select component so that
 * it conforms with the COMATCH styleguide
 *
 * @see https://react-select.com/styles#provided-styles-and-state
 */
export const customStyles = {
    control: () => ({
        border: 'none',
        position: 'relative',
        minWidth: 90,
        zIndex: 1,
    }),
    dropdownIndicator: () => ({
        height: '37px',
        width: 20,
        position: 'absolute',
        right: '.5px',
        top: '.5px',
        // This fix will remove the glossy background for select elements
        // from the Safari browser, while maintaing the default arrow icon
        /* eslint-disable max-len */
        background: `url('data:image/svg+xml;base64,PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA0Ljk1IDEwIj48ZGVmcz48c3R5bGU+LmNscy0xe2ZpbGw6I2ZmZjt9LmNscy0ye2ZpbGw6IzQ0NDt9PC9zdHlsZT48L2RlZnM+PHRpdGxlPmFycm93czwvdGl0bGU+PHJlY3QgY2xhc3M9ImNscy0xIiB3aWR0aD0iNC45NSIgaGVpZ2h0PSIxMCIvPjxwb2x5Z29uIGNsYXNzPSJjbHMtMiIgcG9pbnRzPSIxLjQxIDQuNjcgMi40OCAzLjE4IDMuNTQgNC42NyAxLjQxIDQuNjciLz48cG9seWdvbiBjbGFzcz0iY2xzLTIiIHBvaW50cz0iMy41NCA1LjMzIDIuNDggNi44MiAxLjQxIDUuMzMgMy41NCA1LjMzIi8+PC9zdmc+') no-repeat right 2px center`,
        /* eslint-enable max-len */
        svg: {
            display: 'none',
        },
    }),
    menu: (provided) => ({
        ...provided,
        borderTopLeftRadius: 0,
        borderTopRightRadius: 0,
        borderBottomLeftRadius: 3,
        borderBottomRightRadius: 3,
        marginTop: 0,
    }),
    menuPortal: (provided) => ({
        ...provided,
        zIndex: 4,
    }),
    option: (provided, state) => ({
        ...provided,
        backgroundColor: state.isFocused ? palette.lightGray : palette.white,
        color: state.isFocused ? palette.darkGray : palette.black,
        transition: 'background-color 250ms ease-out',
        '&:hover': {
            backgroundColor: palette.lightGray,
            color: palette.darkGray,
            cursor: 'pointer',
        },
        '&:last-child': {
            borderBottomRightRadius: 3,
            borderBottomLeftRadius: 3,
        },
    }),
    multiValue: () => ({
        border: `1px solid ${palette.gray}`,
        borderRadius: 3,
        color: palette.gray,
        fontSize: typography.fontSm,
        display: 'flex',
        flexDirection: 'row-reverse',
        padding: '0 4px',
        marginRight: 10,
        marginBottom: 3,
        marginTop: 3,
        '&:last-child': {
            marginRight: 0,
        }
    }),
    multiValueLabel: () => ({
        fontSize: typography.fontSm,
        padding: 0,
    }),
    multiValueRemove: () => ({
        fontSize: typography.fontSm,
        paddingRight: 4,
        position: 'relative',
        top: 2,
        '&:hover': {
            cursor: 'pointer',
        },
    }),
    clearIndicator: () => ({
        display: 'none',
    }),
    indicatorSeparator: () => ({
        display: 'none',
    }),
    valueContainer: (provided) => ({
        ...provided,
        padding: '5px 8px',
    }),
};
