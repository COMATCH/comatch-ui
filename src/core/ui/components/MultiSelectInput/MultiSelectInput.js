import React from 'react';
import PropTypes from 'prop-types';
import Select2 from 'react-select';
import classNames from 'classnames';
import noop from 'lodash/noop';
import { InputLabel } from '../InputLabel';
import { InputError } from '../InputError';
import { customStyles } from './custom-styles';

import './MultiSelectInput.scss';

const propTypes = {
    /**
     * Additional class names
     */
    className: PropTypes.string,
    display: PropTypes.oneOf(['block', 'inline']),
    disabled: PropTypes.bool,
    /**
     * HTML id attribute
     */
    id: PropTypes.string,
    inputError: PropTypes.string,
    label: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node, PropTypes.string]),
    name: PropTypes.string.isRequired,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    options: PropTypes.arrayOf(
        PropTypes.shape({
            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
            label: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
        }),
    ).isRequired,
    required: PropTypes.bool,
};

const defaultProps = {
    className: null,
    display: 'block',
    disabled: false,
    id: null,
    inputError: '',
    label: '',
    onBlur: noop,
    onChange: noop,
    required: false,
};

/**
 * An input similar to our `Select` input, allowing the user to select multiple options.
 * 
 * This component is our customized wrapper around the 'react-select' component.
 * @see https://github.com/JedWatson/react-select
 */

export const MultiSelectInput = ({
    className,
    display,
    disabled,
    id,
    inputError,
    label,
    name,
    onBlur,
    onChange,
    options,
    required,
    ...props
}) => {
    const classes = classNames('Input', 'MultiSelectInput', className, display, {
        disabled,
        'has-error': inputError,
    });

    function handleChange(selectedOption) {
        // The 'react-select' by default passes the selected option as an
        // object to the change event. We need to restructure this schema
        // for our MultiSelectInput to conform with our other input components
        onChange({
            target: {
                value: selectedOption,
            },
        });
    }

    return (
        <div id={id} className={classes}>
            {label && <InputLabel text={label} required={required} />}
            <Select2
                isMulti
                className="MultiSelectInput__input"
                isDisabled={disabled}
                name={name}
                onBlur={onBlur}
                onChange={handleChange}
                options={options}
                styles={customStyles}
                {...props}
            />
            {inputError && <InputError text={inputError} />}
        </div>
    );
};

MultiSelectInput.defaultProps = defaultProps;

MultiSelectInput.propTypes = propTypes;
