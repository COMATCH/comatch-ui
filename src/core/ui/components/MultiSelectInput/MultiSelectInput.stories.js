import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { MultiSelectInput } from './MultiSelectInput';

storiesOf('MultiSelectInput', module)
    .add('single with name and options', () => (
        <>
            <MultiSelectInput
                name="fruits"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
            />
            <Highlight className="html">
                {'<MultiSelectInput' +
                    '\n\tname="fruits"' +
                    '\n\toptions={[' +
                    `\n\t\t{ value: 'apple', label: 'Apple' },` +
                    `\n\t\t{ value: 'banana', label: 'Banana' },` +
                    `\n\t\t{ value: 'peach', label: 'Peach' },` +
                    '\n\t]}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('multiple with name options, display inline and display block', () => (
        <div>
            <MultiSelectInput
                name="fruits"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
            />
            <MultiSelectInput
                display="inline"
                name="fruits"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
            />
            <Highlight className="html">
                {'<MultiSelectInput' +
                    '\n\tname="fruits"' +
                    '\n\toptions={[' +
                    `\n\t\t{ value: 'apple', label: 'Apple' },` +
                    `\n\t\t{ value: 'banana', label: 'Banana' },` +
                    `\n\t\t{ value: 'peach', label: 'Peach' },` +
                    '\n\t]}' +
                    '\n/>\n\n'}
                {'<MultiSelectInput' +
                    '\n\tdisplay="inline"' +
                    '\n\tname="fruits"' +
                    '\n\toptions={[' +
                    `\n\t\t{ value: 'apple', label: 'Apple' },` +
                    `\n\t\t{ value: 'banana', label: 'Banana' },` +
                    `\n\t\t{ value: 'peach', label: 'Peach' },` +
                    '\n\t]}' +
                    '\n/>'}
            </Highlight>
        </div>
    ))
    .add('with label', () => (
        <div>
            <MultiSelectInput
                label="MultiSelect your fruit"
                name="fruits"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
            />
            <Highlight className="html">
                {'<MultiSelectInput' +
                    '\n\tlabel="MultiSelect your fruit"' +
                    '\n\tname="fruits"' +
                    '\n\toptions={[' +
                    `\n\t\t{ value: 'apple', label: 'Apple' },` +
                    `\n\t\t{ value: 'banana', label: 'Banana' },` +
                    `\n\t\t{ value: 'peach', label: 'Peach' },` +
                    '\n\t]}' +
                    '\n/>'}
            </Highlight>
        </div>
    ))
    .add('with label and required', () => (
        <div>
            <MultiSelectInput
                label="MultiSelect your fruit"
                name="fruits"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
                required
            />
            <Highlight className="html">
                {'<MultiSelectInput' +
                    '\n\trequired' +
                    '\n\tlabel="MultiSelect your fruit"' +
                    '\n\tname="fruits"' +
                    '\n\toptions={[' +
                    `\n\t\t{ value: 'apple', label: 'Apple' },` +
                    `\n\t\t{ value: 'banana', label: 'Banana' },` +
                    `\n\t\t{ value: 'peach', label: 'Peach' },` +
                    '\n\t]}' +
                    '\n/>'}
            </Highlight>
        </div>
    ))
    .add('with error', () => (
        <div>
            <MultiSelectInput
                inputError="Alarm! Meltdown Imminent!"
                label="MultiSelect your fruit"
                name="fruits"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
                required
            />
            <Highlight className="html">
                {'<MultiSelectInput' +
                    '\n\tinputError="Alarm! Meltdown Imminent!"' +
                    '\n\trequired' +
                    '\n\tlabel="MultiSelect your fruit"' +
                    '\n\tname="fruits"' +
                    '\n\toptions={[' +
                    `\n\t\t{ value: 'apple', label: 'Apple' },` +
                    `\n\t\t{ value: 'banana', label: 'Banana' },` +
                    `\n\t\t{ value: 'peach', label: 'Peach' },` +
                    '\n\t]}' +
                    '\n/>'}
            </Highlight>
        </div>
    ))
    .add('with defaultValue and onChange (uncontrolled)', () => {
        const handleChange = (event) => action('changed')(event.target.name, event.target.value);
        return (
            <div>
                <MultiSelectInput
                    name="fruit"
                    onChange={handleChange}
                    options={[
                        { value: 'apple', label: 'Apple' },
                        { value: 'banana', label: 'Banana' },
                        { value: 'peach', label: 'Peach' },
                    ]}
                    defaultValue={[{ label: 'Peach', value: 'peach' }]}
                />
                <Highlight className="html">
                    {'<MultiSelectInput' +
                        `\n\tdefaultValue={[{ label: 'Peach', value: 'peach' }]}` +
                        '\n\tname="fruits"' +
                        '\n\toptions={[' +
                        `\n\t\t{ value: 'apple', label: 'Apple' },` +
                        `\n\t\t{ value: 'banana', label: 'Banana' },` +
                        `\n\t\t{ value: 'peach', label: 'Peach' },` +
                        '\n\t]}' +
                        '\n/>'}
                </Highlight>
            </div>
        );
    })
    .add('disabled', () => (
        <>
            <MultiSelectInput
                name="fruits"
                disabled
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
            />
            <Highlight className="html">
                {'<MultiSelectInput' +
                    '\n\tdisabled' +
                    '\n\tname="fruits"' +
                    '\n\toptions={[' +
                    `\n\t\t{ value: 'apple', label: 'Apple' },` +
                    `\n\t\t{ value: 'banana', label: 'Banana' },` +
                    `\n\t\t{ value: 'peach', label: 'Peach' },` +
                    '\n\t]}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('as a controlled input', () => (
        <>
            <MultiSelectInputContainer />
            <Highlight className="javascript">
                {`class MultiSelectInputContainer extends React.Component {
    state = { fruit: 'banana' };

    handleChange = (event) => {
        this.setState({ fruit: event.target.value });
        action('changed in container')(event.target.name, event.target.value);
    };

    render() {
        return (
            <MultiSelectInput
                name="fruit"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
                onChange={this.handleChange}
                value={this.state.fruit}
            />
        );
    }
}`}
            </Highlight>
        </>
    ));

class MultiSelectInputContainer extends React.Component {
    state = { fruit: 'banana' };

    handleChange = (event) => {
        this.setState({ fruit: event.target.value });
        action('changed in container')(event.target.name, event.target.value);
    };

    render() {
        return (
            <MultiSelectInput
                name="fruit"
                options={[
                    { value: 'apple', label: 'Apple' },
                    { value: 'banana', label: 'Banana' },
                    { value: 'peach', label: 'Peach' },
                ]}
                onChange={this.handleChange}
                value={this.state.fruit}
            />
        );
    }
}
