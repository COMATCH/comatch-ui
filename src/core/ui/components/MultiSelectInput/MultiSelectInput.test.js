import React from 'react';
import renderer from 'react-test-renderer';
import { MultiSelectInput } from './MultiSelectInput';
import '../../../config/tests/setup';

describe('MultiSelectInput', () => {
    it('should render correctly (without label)', () => {
        const props = {
            name: 'multi-select-input-name',
            options: [
                { value: 'option-1', label: 'Option 1' },
                // { value: 'option-2', label: 'Option 2' },
                // { value: 'option-3', label: 'Option 3' },
            ],
        };
        const multiSelectInput = renderer.create(<MultiSelectInput {...props} />).toJSON();

        expect(multiSelectInput).toMatchSnapshot();
    });

    it('should render correctly (with label)', () => {
        const props = {
            name: 'multi-select-input-name',
            options: [{ value: 'option-1', label: 'Option 1' }],
            label: 'multi-select-input-label',
        };

        const multiSelectInput = renderer.create(<MultiSelectInput {...props} />).toJSON();

        expect(multiSelectInput).toMatchSnapshot();
    });

    it('should render correctly (with label and required *)', () => {
        const props = {
            name: 'multi-select-input-name',
            options: [{ value: 'option-1', label: 'Option 1' }],
            label: 'multi-select-input-label',
            required: true,
        };

        const multiSelectInput = renderer.create(<MultiSelectInput {...props} />).toJSON();

        expect(multiSelectInput).toMatchSnapshot();
    });

    it('should render correctly (with inputError)', () => {
        const props = {
            name: 'multi-select-input-name',
            options: [{ value: 'option-1', label: 'Option 1' }],
            inputError: 'error-sample',
        };
        const multiSelectInput = renderer.create(<MultiSelectInput {...props} />).toJSON();

        expect(multiSelectInput).toMatchSnapshot();
    });
});
