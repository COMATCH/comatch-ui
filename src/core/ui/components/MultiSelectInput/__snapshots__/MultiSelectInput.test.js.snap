// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectInput should render correctly (with inputError) 1`] = `
<div
  className="Input MultiSelectInput block has-error"
  id={null}
>
  
  <div
    className="MultiSelectInput__input css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className=" css-1n4pbi4"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-1hmthh9"
      >
        <div
          className=" css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-5-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-1wy0on6"
      >
        <span
          className=" css-1hyfx7x"
        />
        <div
          aria-hidden="true"
          className=" css-1c73l9k"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <input
        name="multi-select-input-name"
        type="hidden"
      />
    </div>
  </div>
  <span
    className="sc-bwzfXH bZudyR InputError"
  >
    error-sample
  </span>
</div>
`;

exports[`MultiSelectInput should render correctly (with label and required *) 1`] = `
<div
  className="Input MultiSelectInput block"
  id={null}
>
  <label
    className="sc-bdVaJa iThGRf InputLabel"
    htmlFor={null}
    id={null}
  >
    multi-select-input-label
    *
  </label>
  <div
    className="MultiSelectInput__input css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className=" css-1n4pbi4"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-1hmthh9"
      >
        <div
          className=" css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-4-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-1wy0on6"
      >
        <span
          className=" css-1hyfx7x"
        />
        <div
          aria-hidden="true"
          className=" css-1c73l9k"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <input
        name="multi-select-input-name"
        type="hidden"
      />
    </div>
  </div>
  
</div>
`;

exports[`MultiSelectInput should render correctly (with label) 1`] = `
<div
  className="Input MultiSelectInput block"
  id={null}
>
  <label
    className="sc-bdVaJa iThGRf InputLabel"
    htmlFor={null}
    id={null}
  >
    multi-select-input-label
  </label>
  <div
    className="MultiSelectInput__input css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className=" css-1n4pbi4"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-1hmthh9"
      >
        <div
          className=" css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-3-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-1wy0on6"
      >
        <span
          className=" css-1hyfx7x"
        />
        <div
          aria-hidden="true"
          className=" css-1c73l9k"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <input
        name="multi-select-input-name"
        type="hidden"
      />
    </div>
  </div>
  
</div>
`;

exports[`MultiSelectInput should render correctly (without label) 1`] = `
<div
  className="Input MultiSelectInput block"
  id={null}
>
  
  <div
    className="MultiSelectInput__input css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className=" css-1n4pbi4"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-1hmthh9"
      >
        <div
          className=" css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-1wy0on6"
      >
        <span
          className=" css-1hyfx7x"
        />
        <div
          aria-hidden="true"
          className=" css-1c73l9k"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <input
        name="multi-select-input-name"
        type="hidden"
      />
    </div>
  </div>
  
</div>
`;
